src/uct/ib/base/ib_md.c

```diff
@@ -318,9 +321,18 @@ void *uct_ib_md_mem_handle_thread_func(void *arg)
     return UCS_STATUS_PTR(UCS_OK);
 
 err_dereg:
-    for (; mr_idx >= 0; --mr_idx) {
-        uct_ib_dereg_mr(ctx->mrs[mr_idx]);
+    for (i = 0; i < mr_idx; i++) {
+        uct_ib_dereg_mr(ctx->mrs[i]);
+        ctx->mrs[i] = NULL;
+    }
+
+    while (ctx->length > 0) {
+        ctx->mrs[mr_idx] = NULL;
```

Comment:
Finally all `mrs` should be NULL, right?
Maybe just use memset for the whole array?

Reply:
I am assuming that we want to avoid always setting the array to zero, else this would be simpler. @yosefe wdyt?